### PR TITLE
feat: add pricing names and namespaces

### PR DESCRIPTION
**Pull request by noumanahmads**
## Description
Following endpoints are implemented: 
1. /v2/prices/namespaces/{tld}
2. /v2/prices/names/{name}

For details please refer to this [issue](https://github.com/blockstack/stacks-blockchain-api/issues/317)

## Type of Change
- [x] New feature
- [ ] Bug fix
- [x] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
No

## Testing information
Test cases are added for the endpoints implemented.
To run the test cases `npm run test:bns`

## Checklist
- [x] Code is commented where needed
- [ ]  Unit test coverage for new or modified code paths
- [x] `npm run test` passes
- [ ]  Changelog is updated
- [x]  @zone117x for review

### CODE REVIEW

File: src/api/routes/bns/pricing.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    const txOptions: ReadOnlyFunctionOptions = {
+      senderAddress: address,
+      contractAddress: 'ST000000000000000000002AMW42H',
+      contractName: 'bns',
+      functionName: 'compute-namespace-price?',
+      functionArgs: [bufferCVFromString(namespace)],
+      network: stacksNetwork,
+    };
+    console.log(req.params);
```
Comment (zone117x) [inline src/api/routes/bns/pricing.ts:39]: Needs a `// TODO: cleanup`

File: src/api/routes/bns/pricing.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        const response = {
+          units: 'STX',
+          amount: contractCallTx.value.value.toString(10),
+        };
+        res.json(response);
+      } else {
+        res.status(400).json({ error: 'Invalid namespace' });
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (zone117x) [inline src/api/routes/bns/pricing.ts:55]: And here

File: src/tests-bns/api.ts [hunk 3]
```diff
@@ -126,6 +126,26 @@ describe('BNS API', () => {
     expect(query1.status).toBe(200);
   });
```
Comment (zone117x) [inline src/tests-bns/api.ts:128]: Needs a `// TODO: implement schema validation test, implement price check successful test`

File: src/api/routes/bns/pricing.ts (hunk 1 above)
Comment (noumanahmads) [reply src/api/routes/bns/pricing.ts]: removed this log

File: src/api/routes/bns/pricing.ts (hunk 2 above)
Comment (noumanahmads) [reply src/api/routes/bns/pricing.ts]: Removed log

File: src/tests-bns/api.ts (hunk 3 above)
Comment (noumanahmads) [reply src/tests-bns/api.ts:130]: Added TODO's here

**Pull request merged**